src/test/java/org/apache/sling/event/impl/jobs/console/InventoryPluginTest.java

```diff
@@ -0,0 +1,303 @@
+                "\n" +
+                "Topology Capabilities\n" +
+                "No topology information available !\n" +
+                "Scheduled Jobs\n" +
+                "Schedule\n" +
+                "Job Topic< : topic\n" +
+                "Schedules : HOURLY 40, CRON 0 * * * *\n\n\n" +
+                "Active JobQueue: null \n" +
+                "Statistics\n" +
+                "Start Time : 01:00:00:000 1970-Jan-01\n" +
+                "Last Activated : 01:00:00:000 1970-Jan-01\n" +
+                "Last Finished : 01:00:00:000 1970-Jan-01\n" +
+                "Queued Jobs : 0\n" +
+                "Active Jobs : 0\n" +
+                "Jobs : 0\n" +
+                "Finished Jobs : 0\n" +
+                "Failed Jobs : 0\n" +
+                "Cancelled Jobs : 0\n" +
+                "Processed Jobs : 0\n" +
+                "Average Processing Time : -\n" +
+                "Average Waiting Time : -\n" +
+                "Status Info : null\n" +
+                "Configuration\n" +
+                "Type : Ordered\n" +
+                "Topics : \n" +
+                "Max Parallel : 0\n" +
+                "Max Retries : 0\n" +
+                "Retry Delay : 0 ms\n" +
+                "Priority : null\n" +
+                "\n" +
+                "Topic Statistics - null\n" +
+                "Last Activated : 01:00:00:000 1970-Jan-01\n" +
+                "Last Finished : 01:00:00:000 1970-Jan-01\n" +
+                "Finished Jobs : 0\n" +
+                "Failed Jobs : 0\n" +
+                "Cancelled Jobs : 0\n" +
+                "Processed Jobs : 0\n" +
+                "Average Processing Time : -\n" +
+                "Average Waiting Time : -\n" +
+                "\n" +
+                "Apache Sling Job Handling - Job Queue Configurations\n" +
+                "----------------------------------------------------\n" +
+                "Job Queue Configuration: null\n" +
+                "Valid : false\n" +
+                "Type : Ordered\n" +
+                "Topics : \n" +
+                "Max Parallel : 0\n" +
+                "Max Retries : 0\n" +
+                "Retry Delay : 0 ms\n" +
+                "Priority : null\n" +
+                "Ranking : 0\n" +
+                "\n" +
+                "Job Queue Configuration: null\n" +
+                "Valid : false\n" +
+                "Type : Ordered\n" +
+                "Topics : \n" +
+                "Max Parallel : 0\n" +
+                "Max Retries : 0\n" +
+                "Retry Delay : 0 ms\n" +
+                "Priority : null\n" +
+                "Ranking : 0\n\n", output);
+    }
+
+    @Test
+    public void printJsonFormat() {
+        StringWriter writerOutput = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(writerOutput);
+
+        inventoryPlugin.print(printWriter, Format.JSON, false);
+
+        String output = writerOutput.toString();
+        assertEquals("{\n" +
```

Comment:
I was going to say that String comparison is brittle for such a test, and nowadays we have better tools like jsonpath [1] for testing JSON. I think most of this module is really old code from when we didn't use the better tools we have today to generate and test JSON, hence the use of printf to generate JSON I suppose.

However, maybe creating this test was easier in this way if you just captured the output and turned it into a String?

In that case I think that's fine, but it would be nice to also check that the output is valid JSON by parsing it, which I don't think is tested now.

If you want to switch to json path like in [1] you might use a data-driven loop where items have a json path and expected output to verify - happy to discuss the details if desired.

[1] https://github.com/apache/sling-org-apache-sling-graphql-core/blob/a5e684fb476f9c43d44f337a84507a4bebd33ec1/src/test/java/org/apache/sling/graphql/core/engine/DefaultQueryExecutorTest.java#L114

Reply:
> hence the use of printf to generate JSON I suppose 

yes, unfortunately, the usage of printf is inherited from the old code

> In that case I think that's fine, but it would be nice to also check that the output is valid JSON by parsing it, which I don't think is tested now.

This is a good point, particularly given the old inherited code where we are forming JSON manually. I tried to follow the json path suggestion, please let me know your thoughts, and thanks for it.